onnxruntime/core/graph/graph.cc

```diff
@@ -126,9 +127,6 @@ const TensorShapeProto* NodeArg::Shape() const {
 }
 
 void NodeArg::SetShape(const TensorShapeProto& shape) {
-  if (!node_arg_info_.has_type()) {
-    return;
-  }
 
```

Comment:
Why remove this check? Is it valid to make changes if type wasn't previously set?

Reply:
We no longer use_has_type(). The new utils::HasType() would simlpy check if the switch value is not VALUE_NOT_SET which would be a default for value_case and that one is already handled by the below switch() statement. It is OK to call type() if there is no type, it would simply return a default_instance with value_case() being VALUE_NOT_SET.

---
In reply to: [322032683](https://github.com/microsoft/onnxruntime/pull/1778#discussion_r322032683) [](ancestors = 322032683)